repo/dispute_case_record.go

```diff
@@ -99,3 +99,43 @@ func (r *DisputeCaseRecord) Contract() *pb.RicardianContract {
 	}
 	return contract
 }
+
+// ResolutionPaymentFeePerByte returns the preferred outpoints to be used when resolving
+// a pending DisputeCaseResolution based on the provided PayoutRatio
+func (r *DisputeCaseRecord) ResolutionPaymentFeePerByte(ratio PayoutRatio, defaultFee uint64) uint64 {
+	switch {
+	case ratio.BuyerMajority(), ratio.EvenMajority():
+		return r.BuyerContract.BuyerOrder.RefundFee
+	case ratio.VendorMajority():
+		if len(r.VendorContract.VendorOrderFulfillment) > 0 && r.VendorContract.VendorOrderFulfillment[0].Payout != nil {
+			return r.VendorContract.VendorOrderFulfillment[0].Payout.PayoutFeePerByte
+		}
+	}
+	return defaultFee
+}
+
+// ResolutionPaymentOutpoints returns the preferred outpoints to be used when resolving
+// a pending DisputeCaseResolution based on the provided PayoutRatio
+func (r *DisputeCaseRecord) ResolutionPaymentOutpoints(ratio PayoutRatio) (outpoints []*pb.Outpoint) {
+	if ratio.VendorMajority() {
+		outpoints = r.VendorOutpoints
+		if outpoints == nil {
+			outpoints = r.BuyerOutpoints
+		}
+	} else {
+		outpoints = r.BuyerOutpoints
+		if outpoints == nil {
+			outpoints = r.VendorOutpoints
+		}
+	}
+	return
+}
+
+// ResolutionPaymentContract returns the preferred contract to be used when resolving
+// a pending DisputeCaseRecord based on the provided PayoutRatio
+func (r *DisputeCaseRecord) ResolutionPaymentContract(ratio PayoutRatio) (contract *pb.RicardianContract) {
+	if ratio.VendorAny() {
```

Comment:
This should return whoever has the majority rather than returning the vendor contract if he receives any payout at all.